chain-impl-mockchain/src/leadership/mod.rs

```diff
@@ -0,0 +1,83 @@
+use crate::block::SignedBlock;
+use crate::key::PublicKey;
+use crate::update::LeaderSelectionDiff;
+use chain_core::property;
+
+pub mod bft;
+
+#[derive(Debug)]
+pub enum LeaderSelection {
+    BFT(bft::BftLeaderSelection<PublicKey>),
+    Genesis,
+}
+
+#[derive(PartialEq, Eq)]
+pub enum IsLeading {
+    Yes,
+    No,
+}
+
+impl From<bool> for IsLeading {
+    fn from(b: bool) -> Self {
+        if b {
+            IsLeading::Yes
+        } else {
+            IsLeading::No
+        }
+    }
+}
+
+impl property::LeaderSelection for LeaderSelection {
+    type Update = LeaderSelectionDiff;
+    type Block = SignedBlock;
+    type Error = Error;
+
+    fn diff(&self, input: &Self::Block) -> Result<Self::Update, Self::Error> {
+        let mut update = <Self::Update as property::Update>::empty();
+
+        match self {
+            LeaderSelection::BFT(ref bft) => {
+                update.bft = property::LeaderSelection::diff(bft, input).map_err(Error::Bft)?;
+            }
+            LeaderSelection::Genesis => {}
```

Comment:
Should there really be `unimplemented!()` so that it does not silently do nothing when we hit this branch in production?

Reply:
good point